homeassistant/components/switcher_kis/light.py

```diff
@@ -0,0 +1,138 @@
+"""Switcher integration Light platform."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from aioswitcher.api import SwitcherBaseResponse, SwitcherType2Api
+from aioswitcher.device import (
+    DeviceCategory,
+    DeviceState,
+    SwitcherSingleShutterDualLight,
+)
+
+from homeassistant.components.light import ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import SIGNAL_DEVICE_ADD
+from .coordinator import SwitcherDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+API_SET_LIGHT = "set_light"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Switcher light from a config entry."""
+
+    @callback
+    def async_add_light(coordinator: SwitcherDataUpdateCoordinator) -> None:
+        """Add light from Switcher device."""
+        if (
+            coordinator.data.device_type.category
+            == DeviceCategory.SINGLE_SHUTTER_DUAL_LIGHT
+        ):
+            async_add_entities(
+                [
+                    SwitcherLightEntity(coordinator, 0),
+                    SwitcherLightEntity(coordinator, 1),
+                ]
+            )
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, SIGNAL_DEVICE_ADD, async_add_light)
+    )
+
+
+class SwitcherLightEntity(
+    CoordinatorEntity[SwitcherDataUpdateCoordinator], LightEntity
+):
+    """Representation of a Switcher light entity."""
+
+    _attr_color_mode = ColorMode.ONOFF
+    _attr_supported_color_modes = {ColorMode.ONOFF}
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SwitcherDataUpdateCoordinator,
+        light_id: int = 0,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._light_id = light_id
+        self.control_result: bool | None = None
+
+        # Entity class attributes
+        self._attr_name = f"Light{self._light_id + 1}"
```

Comment:
Let's add a translation key and use translation placeholders to add the light id. We should however add a space in between the light and the id

Reply:
I guess this can be resolved now :)